django-backend/fecfiler/web_services/dot_fec_serializer.py

```diff
@@ -0,0 +1,88 @@
+from django.db import models
+from fecfile_validate import validate
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def boolean_serializer(model_instance, field_name):
+    return "X" if getattr(model_instance, field_name, None) else ""
+
+
+def date_serializer(model_instance, field_name):
+    date = getattr(model_instance, field_name, None)
+    return date.strftime("%Y%m%d") if date else ""
+
+
+def foreign_key_serializer(model_instance, field_name):
+    """returns value of foreign key rather than the object it points to
+    For example: we have a foreign key for `report_code`.  django stores the key
+    in `report_code_id` and joins the `report_code_label` row into `report_code`.
+    Because we just want to write the key ("Q1" or something) we have to add "_id"
+    to the field name.
+    """
+    return getattr(model_instance, field_name + "_id", "")
+
+
+def default_serializer(model_instance, field_name):
+    """For most field types, just stringifying the value will work.
+    In the case where the field is None, we want empty string rather than
+    "None", thus the falsy condition
+    """
+    value = getattr(model_instance, field_name)
+    return str(value) if value else ""
+
+
+"""A map of model field types to their serializers.
+Pass the model instance and field name into the serializer to
+get a string representation in the FEC standard
+"""
+FIELD_SERIALIZERS = {
+    models.BooleanField: boolean_serializer,
+    models.DateField: date_serializer,
+    models.ForeignKey: foreign_key_serializer,
+    None: default_serializer,
```

Comment:
RIP lambdas.  They were a very elegant solution, but this is really cool in its own right and should be much, much easier to expand on.

Reply:
yeah i think you were right to point out that they were getting unreadable.